src/typemap.c

```diff
@@ -548,8 +549,8 @@ int jl_typemap_intersection_visitor(union jl_typemap_t map, int offs,
                 else {
                     // else an array scan is required to check subtypes
                     // first, fast-path: optimized pre-intersection test to see if `ty` could intersect with any Type
-                    if (typetype || jl_type_intersection((jl_value_t*)jl_type_type, ty) != jl_bottom_type)
-                        if (!jl_typemap_intersection_array_visitor(&cache->targ, ty, 1, offs, closure)) return 0;
+                    //if (typetype || jl_type_intersection((jl_value_t*)jl_type_type, ty) != jl_bottom_type)
```

Comment:
Seems like it should still work. I don't remember why I disabled it. I'll try it now.